### PR TITLE
DOCSUP-7842: Document the dictionary function

### PR DESCRIPTION
**Pull request by sevirov**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:

- Documentation for #21910

### CODE REVIEW

File: docs/en/sql-reference/table-functions/dictionary.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+---
+toc_priority: 54
+toc_title: dictionary function
```
Comment (olgarev) [inline docs/en/sql-reference/table-functions/dictionary.md:3]:
```suggestion
toc_title: Dictionary functions
```

File: docs/en/sql-reference/table-functions/dictionary.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+┌─id─┬─value─┐
+│  0 │     0 │
+│  1 │     1 │
+└────┴───────┘
+```
+
+Create a dictionary:
+
+``` sql
+CREATE DICTIONARY table_function_dictionary_test_dictionary(id UInt64, value UInt64 DEFAULT 0) PRIMARY KEY id
```
Comment (olgarev) [inline docs/en/sql-reference/table-functions/dictionary.md:38]:
```suggestion
CREATE DICTIONARY table_with_dictionary(id UInt64, value UInt64 DEFAULT 0) PRIMARY KEY id
```

File: docs/en/sql-reference/table-functions/dictionary.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+SOURCE(CLICKHOUSE(HOST 'localhost' PORT tcpPort() USER 'default' TABLE 'table_function_dictionary_source_table')) LAYOUT(DIRECT());
+```
+
+Query:
+
+``` sql
+SELECT * FROM dictionary('table_function_dictionary_test_dictionary');
```
Comment (olgarev) [inline docs/en/sql-reference/table-functions/dictionary.md:45]:
```suggestion
SELECT * FROM dictionary('table_with_dictionary');
```

File: docs/en/sql-reference/table-functions/dictionary.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+│  0 │     0 │
+│  1 │     1 │
+└────┴───────┘
+```
+
+Create a dictionary:
+
+``` sql
+CREATE DICTIONARY table_function_dictionary_test_dictionary(id UInt64, value UInt64 DEFAULT 0) PRIMARY KEY id
+SOURCE(CLICKHOUSE(HOST 'localhost' PORT tcpPort() USER 'default' TABLE 'table_function_dictionary_source_table')) LAYOUT(DIRECT());
```
Comment (olgarev) [inline docs/en/sql-reference/table-functions/dictionary.md:39]:
```suggestion
SOURCE(CLICKHOUSE(HOST 'localhost' PORT tcpPort() USER 'default' TABLE 'dictionary_source_table')) LAYOUT(DIRECT());
```

File: docs/en/sql-reference/table-functions/dictionary.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+-   `dict` — A dictionary name. [String](../../sql-reference/data-types/string.md).
+
+**Returned value**
+
+A ClickHouse table.
+
+**Example**
+
+Input table:
```
Comment (olgarev) [inline docs/en/sql-reference/table-functions/dictionary.md:26]:
```suggestion
Input table `dictionary_source_table`:
```

File: docs/en/sql-reference/table-functions/dictionary.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
+---
+toc_priority: 54
+toc_title: dictionary function
+---
+
+# dictionary {#dictionary-function}
+
+Displays the [dictionary](../../sql-reference/dictionaries/external-dictionaries/external-dicts.md) data as a ClickHouse table.
```
Comment (olgarev) [inline docs/en/sql-reference/table-functions/dictionary.md:8]:
```suggestion
Displays the [dictionary](../../sql-reference/dictionaries/external-dictionaries/external-dicts.md) data as a ClickHouse table. Works the same way as [Dictionary](../../engines/table-engines/special/dictionary.md) engine.
```

**Comment by sevirov** [comment]
@kitaisreal Please, review this PR.

**Review by kitaisreal — APPROVED** [review]

**Pull request merged**